### PR TITLE
Added the code for LCA of BST in Java

### PR DESCRIPTION
**Pull request by Jassi10000-zz**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes

### Description

(Write your answer here.)
Successfully added the code for Least Common Ancestor in Binary Search Tree in Java Language in best time and space complexity

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2639

### CODE REVIEW

File: Java/ds/LCAofaBST.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import java.util.Scanner;
```
Comment (HarshCasper) [inline Java/ds/LCAofaBST.java:8]: Add a beginner pitch

File: Java/ds/LCAofaBST.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// 30 70
+// O/P :
+// 60
+
+// I/P :
+// 6
+// 50 40 70 30 41 69 75
+// 30 41
+// O/P :
+// 50
```
Comment (HarshCasper) [inline Java/ds/LCAofaBST.java]: Put as a multi-line comment

File: Java/ds/LCAofaBST.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    int data;
+
+    Node(int data) {
+        this.data = data;
+        left = null;
+        right = null;
+    }
+}
+
+class LCAofaBST {
```
Comment (HarshCasper) [inline Java/ds/LCAofaBST.java:24]: Add function description

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Java/ds/LCAofaBST.java (hunk 1 above)
Comment (Jassi10000-zz) [reply Java/ds/LCAofaBST.java:8]: can you explain this thing , 
what's meant by beginner pitch

File: Java/ds/LCAofaBST.java (hunk 2 above)
Comment (Jassi10000-zz) [reply Java/ds/LCAofaBST.java]: sure will do that

**Comment by Jassi10000-zz** [comment]
@HarshCasper I have made all the necessary changes asked by you after the review , 
So please review my latest commits.

**Review by Jassi10000-zz — COMMENTED** [review]
I have reviewed my code from my side
@HarshCasper Please have a look and review my latest commits

File: Java/ds/LCAofaBST.java (hunk 1 above)
Comment (ankitaggarwal23) [reply Java/ds/LCAofaBST.java:8]: Beginner friendly description

File: Java/ds/LCAofaBST.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+// 30 70
+// O/P :
+// 60
+
+// I/P :
+// 6
+// 50 40 70 30 41 69 75
+// 30 41
+// O/P :
+// 50
```
Comment (HarshCasper) [inline Java/ds/LCAofaBST.java]: Use Multi-Line Comments here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Jassi10000-zz** [comment]
@HarshCasper  I have made the changes requested , please check once

**Comment by HarshCasper** [comment]
Add a beginner friendly description first. Address the changes requested first

**Comment by Jassi10000-zz** [comment]
@HarshCasper Now check once

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**